### PR TITLE
Add null checks for gazeProviderPointingData

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

This change is needed since the gaze provider isn't a required pointer. Some scenarios need to support the lack of a gaze pointer.

In those cases, this pointing data is never initialized and can null ref in a few spots.

## Changes
- Fixes: #7427

### CODE REVIEW

File: Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs [hunk 1]
```diff
@@ -556,7 +556,7 @@ private void UpdateGazeProvider()
         {
             // The gaze hit result may be populated from previous raycasts this frame, only recompute
             // another raycast if it's not populated
-            if (gazeHitResult == null)
+            if (gazeHitResult == null && gazeProviderPointingData != null)
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: Not sure understanding why this is necessary? Seems like UpdateGazeProvider() won't be called at all if gazeProviderPointingData or it's pointer is null? See L 542 above

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: While that's true at the moment, I figured it'd better be safe than sorry since nothing is preventing this method from being called elsewhere in the script.

Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: Sure, and I guess I would say two things.

1) Remove the gazeProviderPointingData null check that is wrapped around the call to UpdateGazeProvider()

2) Concerning Julia's point, I feel like there are a lot of items that could be null in FocusProvider (as well as of course in almost every component) and adding a null check test for just one doesn't seem like it delivers a lot. It would be better to add test holistically and maybe even for the items we expect likely will be null in some instance

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: I think this is a unique thing in FocusProvider though, where it being null is actually "okay". We want to support the scenario where the gaze provider is optional, and right now that's not possible.

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: Removed the duplicate null check

**Comment by keveleigh** [comment]
@julenka 

>What are your thoughts on adding a test case to verify that gaze provider can be null?

Added a test! I tested it without my fix, and it (expectedly) threw a null ref. With my change, it passes. It tests that FocusProvider can cycle without a gaze provider while also unintentionally tests the case where a gaze provider is destroyed at runtime, to make sure various pieces are cleaned up correctly.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs [hunk 2]
```diff
@@ -384,6 +385,36 @@ public IEnumerator TestPrioritizedLayerMask()
             Assert.IsNull(shellHandRayPointer.Result?.CurrentPointerTarget, $"{noPriorityCube.name} should NOT be raycast target by shell hand ray pointer");
         }
 
+        /// <summary>
+        /// </summary>
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs]: please describe what you're testing in the summary

File: Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs [hunk 3]
```diff
@@ -558,19 +553,26 @@ private void UpdateGazeProvider()
             // another raycast if it's not populated
             if (gazeHitResult == null)
             {
-                // get 3d hit
-                hitResult3d.Clear();
-                var raycastProvider = CoreServices.InputSystem.RaycastProvider;
-                LayerMask[] prioritizedLayerMasks = (gazeProviderPointingData.Pointer.PrioritizedLayerMasksOverride ?? FocusLayerMasks);
-                QueryScene(gazeProviderPointingData.Pointer, raycastProvider, prioritizedLayerMasks,
-                    hitResult3d, maxQuerySceneResults, focusIndividualCompoundCollider);
+                if (gazeProviderPointingData != null)
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: wouldn't we have to check for gazeProviderPointingData.Pointer != null as well?

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs]: Ah good call!

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs (hunk 2 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs]: Oh weird, i could have sworn I filled this out! I'll update
